tests/test_basic_s3.py

```diff
@@ -572,9 +641,13 @@ async def test_presign_sigv4(
     ), msg
 
     # Try to retrieve the object using the presigned url.
-    async with aio_session.get(presigned_url) as resp:
-        data = await resp.read()
-        assert data == b'foo'
+    if httpx and isinstance(aio_session, httpx.AsyncClient):
+        async with aio_session.stream("GET", presigned_url) as resp:
+            data = await resp.aread()
```

Comment:
what do you think of making an adapter class so no test changes are necessary.  We can expose the raw stream via a new property if needed

Reply:
*no* test changes is not going to be possible, but I can try to make one that implements a bunch of the basic functionality which would minimize test changes.

Reply:
well, translating `await resp.read()` into `await resp.aread()` with an adapter class is trivial - so this test case can be fixed.

But going from `await resp['Body'].aclose` to `resp['Body'].close()`...
I guess it's possible in theory to hook into the currently running async framework and run `.aclose()` from a sync `.close()` in an adapter class... but that feels like a very bad idea. Especially as we're looking to support anyio and structured concurrency.

I suppose I could write a wrapper class that .. only translates `read`->`aread`, and gives specific errors for the other ones? It could maybe help transitioning code currently written, but I think perhaps more appropriate is to if/when dropping aiohttp we make the adapter class raise `DeprecationError` if calling `.read()` or `.close()`.